contracts/contracts/eosio.system/src/delegate_bandwidth.cpp

```diff
@@ -295,6 +297,13 @@ namespace eosiosystem {
       check( _gstate.total_activated_stake >= min_activated_stake,
              "cannot undelegate bandwidth until the chain is activated (at least 15% of all tokens participate in voting)" );
 
+      if(is_block_producer(from))
+      {
+          del_bandwidth_table     del_tbl( _self, from.value );
+          const auto& del = del_tbl.get( from.value, "user has no resources" );
+          check( current_time_point() > del.staked_time + stake_lock_period, "producer cannot undelegate bandwidth during 180 days");
```

Comment:
In case a BP stakes more tokens during the initial locking period, it will increase the locking proportionally. For example, an account opens as BP with 100M tokens staked with the initial six month locking period. After 1.5 months they stake an additional 200M tokens. As a result, the locking period is adjusted from the remaining 4.5 months to 5.5 months with a total of 300M tokens staked. 6mon - 6mon*(100M*1.5/6)/(100M+200M) = 5.5mon.

Reply:
@roman-tik we will implement this logic but in scope of other PR